### PR TITLE
Support default font_features from fontconfig

### PR DESCRIPTION
**Pull request by Diaoul**
As discussed here: https://github.com/kovidgoyal/kitty/pull/2248

### CODE REVIEW

**Comment by Diaoul** [comment]
I'll do a pass on the doc when I get the validation of the implementation :memo: 

For reference:
```bash
cat ~/.config/fontconfig/fonts.conf
```
```xml
<?xml version="1.0"?>
<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
<fontconfig>
	<alias>
		<family>monospace</family>
		<prefer>
			<family>Fira Code</family>
		</prefer>
	</alias>
  <match target="font">
    <test name="family">
      <string>Fira Code</string>
    </test>
    <edit name="fontfeatures" mode="append">
      <string>ss01</string>
    </edit>
  </match>
</fontconfig>
```

**Comment by kovidgoyal** [comment]
Looks generally fine with a quick glance over. Add the docs in config_data.py and I will review in detail.

**Comment by Diaoul** [comment]
Actually there is a bug in `pattern_as_dict`: it does not handle multiple values correctly. This applies to `FC_FONT_FEATURES` but maybe others as well. For reference, [here](https://gitlab.freedesktop.org/fontconfig/fontconfig/-/blob/master/src/fcdbg.c#L188) is how multiple values are handled by `FcPatternPrint` using [`FcValueListPrint`](https://gitlab.freedesktop.org/fontconfig/fontconfig/-/blob/master/src/fcdbg.c#L129).

**Comment by kovidgoyal** [comment]
Feel free to send a separate PR to fix it or wait and I will get around
to it when I have a moment.

**Comment by kovidgoyal** [comment]
Also, what's the bug, as far as I know, patter_as_dict doesnt get any non-scalar values currently.

**Comment by Diaoul** [comment]
I just fixed it for font_features. As for the others I don't know if they are lists or not and I could not find a reliable source of information for this. [Here](https://cgit.freedesktop.org/fontconfig/tree/fontconfig/fontconfig.h#n123) FC_FONT_FEATURES is defined as a String...

I've seen references to fonts being able to be part of multiple families somewhere (sorry lost the reference) and despite `family` being singular, [the description](https://gitlab.freedesktop.org/fontconfig/fontconfig/-/blob/master/doc/fontconfig-devel.sgml#L138) reads *Font family name**s***. :man_shrugging:

File: kitty/fontconfig.c [hunk 1]
```diff
@@ -37,7 +37,21 @@ pattern_as_dict(FcPattern *pat) {
...
+    PyObject *l = PyList_New(0); \
+    while (get(pat, which, n, &out) == FcResultMatch) { \
+        p = conv(out); if (p == NULL) { Py_CLEAR(l); Py_CLEAR(ans); return NULL; } \
+        if (PyList_Append(l, p) != 0) { Py_CLEAR(p); Py_CLEAR(l); Py_CLEAR(ans); return NULL; } \
+        Py_CLEAR(p); \
+        n++; \
+    } \
+    if (PyDict_SetItemString(ans, #name, l) != 0) { Py_CLEAR(l); Py_CLEAR(ans); return NULL; } \
+    Py_CLEAR(l); \
+}
```
Comment (Diaoul) [inline kitty/fontconfig.c:52]: Please doublecheck that one @kovidgoyal, I tried to make sure I've cleaned up properly but I might have missed something.

**Pull request merged**